### ISSUE TITLE
MOF Compiler recovery from CreateInstance Error

### ISSUE THREAD

**Issue by KSchopmeyer** [pywbem/pywbem#1894]
The MOF compiler  has an issue in its recovery from the condition where it attempts to create an instance and that creation fails.  As documented in issue #1890, it goes through a process of trying to write the instance by using modify instance, etc.  However, if the exception from the original CreateInstance has any status except CIM_ERR_ALREADY_EXISTS it simply drops the whole thing on the floor and does not raise any exception at all.  Since within the MOFWBEMConnection class, the ModifyInstance returns CIM_ERR_FAILED, no instance gets created and no error is returned.   This is never seen in a standard compile because the CreateInstance does not test for dup but simply appends the new instance to its list of instances.

It should then reraise the original exception.

Here is proposed code change

```
            except CIMError as ce2:                                          <==============new name
                if ce2.status_code == CIM_ERR_NOT_SUPPORTED:
                    if p.parser.verbose:
                        p.parser.log(
                            _format("ModifyInstance not supported. "
                                    "Deleting instance of {0!A}: {1}",
                                    inst.classname, inst.path))
                    p.parser.handle.DeleteInstance(inst.path)
                    if p.parser.verbose:
                        p.parser.log(
                            _format("Creating instance of {0!A}.",
                                    inst.classname))
                    p.parser.handle.CreateInstance(inst)
                else:  # modify failed, output original error   <===========reraise original error
                    raise ce
```

**Comment by andy-maier** [comment]
That proposed solution looks like the change in PR #1895. Please clarify whether PR #1895 addresses this issue.

**Comment by andy-maier** [comment]
Karl said that PR #1895 actually fixes this issue here, and issue #1890 was about discussing the approach. Closing this issue here since PR #1895 was merged.

**Comment by andy-maier** [comment]
Fixed by PR #1895 for master, and PR #1897 for 0.15.0.
